pkg/autodiscovery/autoconfig.go

```diff
@@ -469,15 +499,78 @@ func (ac *AutoConfig) pollConfigs() {
 func (ac *AutoConfig) processRemovedConfigs(configs []integration.Config) {
 	ac.scheduler.Unschedule(configs)
 	for _, c := range configs {
-		delete(ac.configResolver.configToService, c.Digest())
 		ac.store.removeLoadedConfig(c)
 		// if the config is a template, remove it from the cache
 		if c.IsTemplate() {
-			ac.templateCache.Del(c)
+			ac.store.templateCache.Del(c)
 		}
 	}
 }
 
+// resolveTemplate attempts to resolve a configuration template using the AD
+// identifiers in the `integration.Config` struct to match a Service.
+//
+// The function might return more than one configuration for a single template,
+// for example when the `ad_identifiers` section of a config.yaml file contains
+// multiple entries, or when more than one Service has the same identifier,
+// e.g. 'redis'.
+//
+// The function might return an empty list in the case the configuration has a
+// list of Autodiscovery identifiers for services that are unknown to the
+// resolver at this moment.
+func (ac *AutoConfig) resolveTemplate(tpl integration.Config) []integration.Config {
```

Comment:
I feel like all the methods starting with `resolve` should be moved to the `config resolver`, WDYT ? I don't clear responsibilities here.

Reply:
so it's mostly wrapping around calling `ConfigResolver.resolve` but I agree we could probably do something better with the naming. Current state:

Autoconfig:
* `resolve` (We could probably rename it to something like `storeAndResolve`)
* `resolveTemplate` (will loop and call `resolveTemplateForService
`)
* `resolveTemplateForService` (wrap `ConfigResolver.resolve` with store access)

`ConfigResolver`:
* resolve